File: コーディングによる問題解決/03 アルゴリズム 初級/全探索/03_整数Vの場所/3.rb
N, V = gets.split.map(&:to_i)
A = gets.split.map(&:to_i)

flag = false
ans = -1

A.each_with_index do |x, i|
    if x == V
        ans = i
        flag = true
    end
end

if flag
    puts ans
else
    puts -1
end

File: コーディングによる問題解決/03 アルゴリズム 初級/全探索/03_整数Vの場所/3.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int V = sc.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = sc.nextInt();
        }

        boolean flag = false;
        int ans = -1;

        for (int i = 0; i < N; i++){
            if (A[i] == V){
                ans = i;
                flag = true;
            }
        }

        if (flag){
            System.out.println(ans);
        }else{
            System.out.println(-1);
        }
    }
}